components/clarity-repl/src/repl/tracer.rs

```diff
@@ -156,8 +156,8 @@ impl EvalHook for Tracer {
             .flat_map(|b| &b.events)
             .collect::<Vec<_>>();
         if emitted_events.len() > self.emitted_events {
-            for i in self.emitted_events..emitted_events.len() {
-                let event = emitted_events[i];
+            for event in emitted_events.iter().skip(self.emitted_events) {
+                // let event = emitted_events[i];
```

Comment:
Readability is not improved in my opinion, but if we're fine with that, let's get rid fo line 160?

Reply:
Agree on readability.

I disabled the lint rule `clippy::needless_range_loop` in a few places where it made things worth.
This one didn't seems to make a big difference to me (both version are a bit difficult)

I renamed `emitted_events` into `nb_of_emitted_events` to make it a bit better